Tests/SwiftSyntaxMacroExpansionTest/MacroSystemTests.swift

```diff
@@ -284,9 +284,37 @@ public struct AddCompletionHandler: PeerMacro {
 
     // This only makes sense for async functions.
     if funcDecl.signature.effectSpecifiers?.asyncSpecifier == nil {
-      throw MacroExpansionErrorMessage(
-        "@addCompletionHandler requires an async function"
+      let newEffects: FunctionEffectSpecifiersSyntax
+      if let existingEffects = funcDecl.signature.effectSpecifiers {
+        newEffects = existingEffects.with(\.asyncSpecifier, .keyword(.async))
+      } else {
+        newEffects = FunctionEffectSpecifiersSyntax(asyncSpecifier: .keyword(.async))
+      }
+
+      let newSignature = funcDecl.signature.with(\.effectSpecifiers, newEffects)
+
+      let diag = Diagnostic(
+        node: Syntax(funcDecl.funcKeyword),
+        message: MacroExpansionErrorMessage(
+          "can only add a completion-handler variant to an 'async' function"
+        ),
+        fixIts: [
+          FixIt(
+            message: MacroExpansionFixItMessage(
+              "add 'async'"
+            ),
+            changes: [
+              FixIt.Change.replace(
+                oldNode: Syntax(funcDecl.signature),
```

Comment:
I don't understand why. 
But when doing this the node.id changes. 

When debugging it first get 
```
▿ SyntaxIdentifier
  - rootId : 4320226936
  ▿ indexInTree : SyntaxIndexInTree
    - indexInTree : 8
```

And when printing inside the FixItApplier I get this
```
▿ SyntaxIdentifier
  - rootId : 105553162409640
  ▿ indexInTree : SyntaxIndexInTree
    - indexInTree : 7
```